lib/tutorials/caching.md

```diff
@@ -0,0 +1,187 @@
+
+### Cache-Control
+
+`Cache-Control` header tells browser and any intermediate cache if resource is cacheable and for how long. For example `Cache-Control:max-age=30, must-revalidate, private` means that browser can cache resource for 30 seconds and `private` means that it should not be cached by intermediate caches, only by browser. `must-revalidate` means that once it expires it has to request the resource again.
+
+Lets see how we can set this header in hapi:
+
+```javascript
+server.route({
+   path: '/hapi/{ttl?}',
+   method: 'GET',
+   handler: function (request, reply) {
+
+      var response = reply({be: 'hapi'});
+      if (request.params.ttl) {
+         response.ttl(request.params.ttl);
+      }
+   },
+   config: {
+      cache: {
+         expiresIn: 30 * 1000,
+         privacy: 'private'
+      }
+   }
+});
+```
+
+Example also illustrates that `expiresIn` value can be overridden with `ttl(msec)` function using [response object](http://hapijs.com/api#response-object) interface.
+
+See [route-options](http://hapijs.com/api#route-options) for more options.
+### Last-Modified
+
+In some cases server can provide information when resources was last modified. Good example is [inert](https://github.com/hapijs/inert) module, which is responsible for serving static files in hapi (`file` and `directory` handler). Last modification timestamp is read for every requested file and send as `Last-Modified` header.
+
+When `Last-Modified` is set on response hapi compares it with `If-Modified-Since` coming from client to decide if should return `304`.
+
+
+Assuming `lastModified` is Date object you can set this header via [response object](http://hapijs.com/api#response-object) interface.
+
+```
+   reply(result).header('Last-Modified', lastModified.toUTCString());
+```
+
+There is one more example using `Last-Modified` in [last section](#client-and-server-caching).
+
+### ETag
+
+ETag is alternative to `last-modified` where server provides token (usually checksum of resource) instead of last modified timestamp. Browser in next request sends this ETag in `If-None-Match` header. Server compares it with new `ETag` and return `304` if it's the same.
+
+You only need to set `ETag` in your handler via `etag(tag, options)` function:
+
+```
+   reply(result).etag('xxxxxxxxx');
+```
+
+Check `options` parameter for `etag` function on [api](http://hapijs.com/api#response-object) page for more details.
+
+
+## Server side caching
+
+hapi provides powerful server side caching via [catbox](https://www.github.com/hapijs/catbox) and makes using cache very convenient. This tutorial help you to understand how catbox is utilized inside hapi and how you can use it. Catbox also can be used out of the hapi, but that is not covered here.
```

Comment:
Maybe remove the sentence about using catbox outside of hapi. Doesn't really seem helpful here.